src/VisualStudio/Core/Def/Implementation/ProjectSystem/CompilationOutputFilesWithImplicitPdbPath.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection.PortableExecutable;
+using Microsoft.CodeAnalysis.Emit;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
+{
+    /// <summary>
+    /// Provides access to compilation outputs based only on the path of the output asssembly.
+    /// If PDB path is known upfront use <see cref="CompilationOutputFiles"/> instead.
+    /// </summary>
+    internal sealed class CompilationOutputFilesWithImplicitPdbPath : CompilationOutputs
+    {
+        public string AssemblyFilePath { get; }
+
+        public CompilationOutputFilesWithImplicitPdbPath(string assemblyFilePath = null)
+        {
+            if (assemblyFilePath != null)
+            {
+                CompilerPathUtilities.RequireAbsolutePath(assemblyFilePath, nameof(assemblyFilePath));
+            }
+
+            AssemblyFilePath = assemblyFilePath;
+        }
+
+        public override string AssemblyDisplayPath => AssemblyFilePath;
+        public override string PdbDisplayPath => Path.GetFileNameWithoutExtension(AssemblyFilePath) + ".pdb";
```

Comment:
No ChangeExtension? I'm guessing there's some evil about it?

Reply:
Does not strip the directory name. I do not want to include the directory since it's unknown where the PDB is gonna be located until we read the assembly, which I don't want to do here. It's just for an error message.